src/timer/timer.c

```diff
@@ -21,34 +21,55 @@
 
 
 #include "timer.h"
+#include "timer_config.h"
 #include "timer_device.h"
 #include "filter.h"
 
-#define QUIET_PERIOD_US	4500
+/* Adds 25% to the max RPM value */
+#define MAX_RPM_MULT	1.25
+#define SEC_IN_A_MIN	60.0
+#define SEC_IN_A_SEC	1.0
+#define US_IN_A_SEC	1000000
 
 static Filter g_timer_filter[CONFIG_TIMER_CHANNELS];
 
-static uint16_t get_timer_quiet_period(const TimerConfig *tc)
+/**
+ * Calculates the highest quiet period usable based on the timer
+ * configurations.  We calculate this by figuring out the expected
+ * maximum HZ of the engine based in the info provided.  Then we simply
+ * take the inverse of that and multiply it by the number of micro seconds
+ * in a second.
+ */
+static unsigned int get_timer_auto_quiet_period(const TimerConfig *tc,
+                                                const float period)
 {
-        /* Only use filtering when in RPM mode */
-        if (MODE_LOGGING_TIMER_RPM != tc->mode)
-                return 0;
-
-        /*
-         * Software Filter Hack Represents 15K Max.  For release
-         * in 2.8.8.  Have to divide value by pulses per revolution
-         * since this must scale depending on the number of pulses
-         * per engine rotation.
-         */
-        return QUIET_PERIOD_US / tc->pulsePerRevolution;
+        const float max_rpm = tc->cfg.max;
+        const float ppr = (float) tc->pulsePerRevolution;
+        const float max_hz = max_rpm * ppr * MAX_RPM_MULT / period;
+
+        return max_hz ? US_IN_A_SEC / max_hz : 0;
+}
+
+static unsigned int get_timer_quiet_period(const TimerConfig *tc)
```

Comment:
For static methods I think we don't need the `timer_` prefix.  It is unnecessary since no-one will be using the method outside the scope of this file.  Hence I think its acceptable to simply name this `get_quiet_period`.  I will name the above `calc_quiet_period`.  It simple that way and easy to follow